aa-sdk/core/src/middleware/erc7677middleware.ts

```diff
@@ -154,8 +156,11 @@ export function erc7677Middleware<
       ...uo,
       paymaster,
       paymasterData,
-      paymasterPostOpGasLimit,
-      paymasterVerificationGasLimit,
+      // optional fields
+      ...(paymasterVerificationGasLimit
+        ? { paymasterVerificationGasLimit }
+        : {}),
+      ...(paymasterPostOpGasLimit ? { paymasterPostOpGasLimit } : {}),
```

Comment:
Is this not equivalent to before?

Reply:
Okay now I understand - the difference between this logic and just adding `paymasterVerificationGasLimit` and `paymasterPostOpGasLimit` directly is that it won't overwrite an existing value if the returned value is `undefined`, but it will overwrite if its defined.

What was happening was `paymasterVerificationGasLimit` was overriding the value that was already estimated and putting in `undefined